### DISCUSSION TITLE
Jest integration

### DISCUSSION THREAD

**Discussion by fulgore83** [mikro-orm/mikro-orm#1493] [General]
Hello, 

I write tests in JEST framework and i use entities here. 
My entities uses decorators and when i call **add** method on _Collection_, orm mechanism required metadata. Then i got error like this:

`TypeError: Cannot read property 'properties' of undefined`

So could You please tell me how can i configure jest and what i shoud trigger from Mikro ORM? I dont want use MikroORM.init because there a need use db data. I dont want use db factor in this tests.

That i think i need use call somehow _ReflectMetadataProvider_ with my entities in discovery process to parsing metadata correctly before using this entities in tests.

Please help me :)
Best regards

**Comment by B4nan** [comment]
You need to use `MikroORM.init` so the metadata discovery happens. If you do not need the actual db connection, use the second parameter that allows you to not connect.

**Reply by fulgore83** [reply]
thank You for that really fast answer, I was thought is there any other way but if not i will do it exactly like You wrote.

Thank You again

**Reply by B4nan** [reply]
if you look at the implementation of `init` method, it will not do much more than the configuration setup and discovery

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/MikroORM.ts#L25

i guess we should catch that error you mentioned and throw something more informative in case the discovery did not happen yet
